airflow/providers/amazon/aws/hooks/ecr.py

```diff
@@ -0,0 +1,100 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import base64
+from dataclasses import dataclass
+from datetime import datetime
+
+from airflow.compat.functools import cached_property
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.log.secrets_masker import mask_secret
+
+
+@dataclass
+class EcrCredentials(LoggingMixin):
+    """Helper class for store temporary ECR credentials."""
+
+    username: str
+    password: str
+    proxy_endpoint: str
+    expires_at: datetime
+
+    def __post_init__(self):
+        mask_secret(self.password)
+        self.log.debug("Credentials to Amazon ECR %r expires at %s.", self.proxy_endpoint, self.expires_at)
+
+    @cached_property
+    def registry(self) -> str:
+        """Return registry in appropriate `docker login` format."""
+        # https://github.com/docker/docker-py/issues/2256#issuecomment-824940506
+        return self.proxy_endpoint.replace("https://", "")
```

Comment:
proxy_endpoint is not immutable. Are you sure cached_property is a good idea ?

Reply:
Ooops... Yep initially I make this dataclass frozen (kind of immutable) however after I add inheritance from `LoggingMixin` so it can't be anymore frozen and forget about this property.

Reply:
And also there is no point to make it cacheable - no heavy computing here